src/xcode/ENA/ENA/Source/Scenes/TraceLocations/Details/TraceLocationDetailsViewController.swift

```diff
@@ -58,25 +58,33 @@ class TraceLocationDetailsViewController: UIViewController, FooterViewHandling {
 			onDuplicateButtonTap(viewModel.traceLocation)
 		}
 	}
-	
+
+	// MARK: - Private
+
+	private let viewModel: TraceLocationDetailsViewModel
+
+	private let onPrintVersionButtonTap: (PDFView) -> Void
+	private let onDuplicateButtonTap: (TraceLocation) -> Void
+	private let onDismiss: () -> Void
+	private var subscriptions = [AnyCancellable]()
+
 	private func generateAndPassQRCodePoster() {
 		viewModel.fetchQRCodePosterTemplateData { [weak self] templateData in
 			switch templateData {
 			case let .success(templateData):
 				DispatchQueue.main.async { [weak self] in
 					self?.footerView?.setLoadingIndicator(false, disable: false, button: .primary)
-
 					do {
 						let pdfView = try self?.createPdfView(templateData: templateData)
 						self?.onPrintVersionButtonTap(pdfView ?? PDFView())
 					} catch {
+						self?.footerView?.setLoadingIndicator(false, disable: false, button: .primary)
 						Log.error("Could not create the PDF view.", log: .qrCode, error: error)
 					}
 				}
 			case let .failure(error):
-					self?.footerView?.setLoadingIndicator(false, disable: false, button: .primary)
-				
-				Log.error("Could not retrieve QR code poster template from protobuf.", log: .qrCode, error: error)
+				self?.footerView?.setLoadingIndicator(false, disable: false, button: .primary)
```

Comment:
Same as above

Reply:
We need this one for failure block, I will remove the one on line 81.